stdlib/public/Distributed/DistributedActorSystem.swift

```diff
@@ -555,9 +555,12 @@ extension DistributedActorSystem {
         errorCode: .typeDeserializationFailure)
     }
 
+    var executeDistributedTargetHasThrown = false
     func doDestroyReturnTypeBuffer<R>(_: R.Type) {
       let buf = resultBuffer.assumingMemoryBound(to: R.self)
-      buf.deinitialize(count: 1)
+      if R.self != Void.self && !executeDistributedTargetHasThrown {
```

Comment:
I actually wonder whether this is reliable, technically we could put result into the buffer and throw afterwards?

Reply:
Hmmm, would we...? Maybe... I'll be digging deeper into this today, could not reproduce the issue so far. But definitely seems like on some platforms may be the not initialzied memory then getting deinitialized makes us crash, so there has to be a way for us to know if to deinit or not hmm

Reply:
One idea I had is to set a bit on the buffer pointer itself to indicate that was initialized, that way even if something throws we'd still de-initialize the memory.